src/worker/sandbox/cgroups/cgroup.go

```diff
@@ -20,10 +22,41 @@ type CgroupImpl struct {
 	memLimitMB int
 }
 
-func (cg *CgroupImpl) printf(format string, args ...any) {
+func (cg *CgroupImpl) printf(level string, format string, args ...any) {
 	if common.Conf.Trace.Cgroups {
+
 		msg := fmt.Sprintf(format, args...)
-		log.Printf("%s [CGROUP %s: %s]", strings.TrimRight(msg, "\n"), cg.pool.Name, cg.name)
+		currLevel := common.Conf.Log.CgroupsLevel
+		logger := slog.New(slog.Default().Handler())
+
+		if (common.Conf.Log.LogFormat == "text") {
+			logFilePath := path.Join(common.Conf.Log.Log_file_dir, "log.txt")
+			f, err := os.OpenFile(logFilePath, os.O_RDWR|os.O_APPEND, 0666)
+			if err != nil {
+				panic(fmt.Errorf("Cannot open log file at %s", logFilePath))
+			}
+			logger = slog.New(slog.NewTextHandler(f, nil))
+			//defer f.Close()
+		} else if (common.Conf.Log.LogFormat == "json") {
+			logFilePath := path.Join(common.Conf.Log.Log_file_dir, "log.json")
+			f, err := os.OpenFile(logFilePath, os.O_RDWR|os.O_APPEND, 0666)
+			if err != nil {
+				panic(fmt.Errorf("Cannot open log file at %s", logFilePath))
+			}
+			logger = slog.New(slog.NewJSONHandler(f, nil))
+			//defer f.Close()
+		}
+
+		if currLevel == "INFO" && level == currLevel {
```

Comment:
The only time levels should potentially appear as strings in the code is when we're reading from the config, everywhere else we should use the different .Info, .Warn functions, etc. OR we should use Level objects: https://pkg.go.dev/log/slog#Level